frontend/app/src/components/RotationChallenge.svelte

```diff
@@ -0,0 +1,191 @@
+<script lang="ts">
+    import { onMount } from "svelte";
+    import Overlay from "./Overlay.svelte";
+    import ModalContent from "./ModalContent.svelte";
+    import Button from "./Button.svelte";
+    import ButtonGroup from "./ButtonGroup.svelte";
+    import Translatable from "./Translatable.svelte";
+    import { i18nKey } from "@src/i18n/i18n";
+    import { mobileWidth } from "@src/stores/screenDimensions";
+
+    const TOLERANCE = 0.1; // tolerance in radians
+
+    interface Props {
+        onClose: () => void;
+        onResult: (e: MouseEvent, success: boolean) => void;
+        imagePath?: string;
+    }
+
+    let { onClose, onResult, imagePath = "/assets/robot.svg" }: Props = $props();
+
+    let leftImage: HTMLImageElement;
+    let targetRotation: number = $state(0);
```

Comment:
Should this just be `const targetRotation = normaliseAngle(Math.random() * 2 * Math.PI)`?
Because it should never change, right?